bin.src/thresholdFits.py

```diff
@@ -0,0 +1,271 @@
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import os
+import argparse
+from glob import glob
+import astropy.io.fits as fits
+
+
+def expand_files(filename_list=None):
+    """Expand a set of filenames and globs, returning a list of filenames
+
+    Parameters
+    ------------
+    filename_list:  list
+        A list of files and glob patterns
+
+    Returns:
+    -----------
+    expandedList : list
+        An expanded list of files that exist and files that fit the glob pattern
+    """
+    expanded_list = []
+    for globPattern in filename_list:
+        files = glob(globPattern)
+
+        if not files:
+            print("%s doesn't match any file" % globPattern)
+            continue
+
+        expanded_list.extend(files)
+
+    return expanded_list
+
+
+def get_size(filepath=None):
+    """
+    Convenience function to return the filesize.
+
+    Parameters:
+    -----------
+    filepath : str
+        Path to the file to check
+
+    Returns:
+    --------
+    size : str
+        Human-readable filesize (Kilobytes or Megabytes)
+    """
+    size_bytes = os.path.getsize(filepath)
+    if size_bytes < (1024 * 1024):
+        size = str(round(size_bytes / 1024)) + "K"
+    else:
+        size = str(round((size_bytes / (1024 * 1024)), 1)) + "M"
+    return size
+
+
+def threshold_file(
+    fits_file=None,
+    threshold=None,
+    not_binary=False,
+    subtract_threshold=False,
+    keep_original_value=True,
+    out_dir=None,
+):
+    """
+    Threshold a FITS file, by changing the value of pixels for
+    each .data element. By default, pixels below the threshold
+    are set to 0, and those above to 1. The headers are kept
+    unchanged.
+
+    Parameters:
+    -----------
+    fits_file : str
+        Path to the FITS file to threshold
+    threshold : float, optional
+        A value of threshold to apply
+    not_binary: bool, optional
+        If set to True, do non-binary thresholding.
+    subtract_threshold: bool, optional
+        If set to True, set pixels below threshold to 0,
+        from those above, subtract the threshold
+    keep_original_value: bool, optional
+        If set to True, set pixels below threshold to 0,
+        keep the value of those above threshold
```

Comment:
In which case we will use this option?

Reply:
`keep_original_value` is the opposite of `subtract_threshold`, i.e. if the user chooses `--not_binary  --subtract_threshold` we subtract the threshold. I could do it so that by default in `--not_binary` we do not subtract threshold, equivalent to `--keep_original_value`. Then the  relevant lines would be 

`if subtract_threshold:
print("subtracting threshold")
    thresh_data[~below_thresh] = raw_data[~below_thresh] - threshold

# do nothing
if keep_original_value:
    print("keeping original value ")
    pass`

Reply:
Sorry to let you misunderstand. My question is not the code itself but the possible use case. We could keep the original code and if we do not feel we will use this option in the real simulation/study/research, we could remove this in a latter time. I just try to understand in which scenario we may want to use this option. Thanks!

Reply:
Ok, I see.  The intention was to explicitly say "I want `--not_binary` and `--keep_original_value`". But I could explain that with "`--not_binary`, by default we keep the original value".  This hopefully makes it simpler.

Reply:
To simplify the code I've removed `-keep_original_value` and explained that this is the default behavior when `--not_binary` flag is used. To subtract the threshold, set `--subtract_threshold`.